geode-core/src/main/java/org/apache/geode/internal/cache/DistributedCacheOperation.java

```diff
@@ -415,15 +415,17 @@ protected void _distribute() {
       if (recipients.isEmpty() && adjunctRecipients.isEmpty() && needsOldValueInCacheOp.isEmpty()
           && cachelessNodes.isEmpty()) {
         if (region.isInternalRegion()) {
-          if (mgr.getNormalDistributionManagerIds().size() > 1) {
-            // suppress this msg if we are the only member.
-            if (logger.isTraceEnabled()) {
-              logger.trace("<No Recipients> {}", this);
-            }
-          } else {
-            // suppress this msg if we are the only member.
-            if (logger.isDebugEnabled()) {
-              logger.debug("<No Recipients> {}", this);
+          if (logger.isDebugEnabled() || logger.isTraceEnabled()) {
```

Comment:
There are actually a few of checks for `logger.isDebugEnabled()` and `logger.isTraceEnabled()` in this method. Maybe we should check it once and stash in local variable. All these redundant checks are going to add up in this method.

Reply:
You might be right that these checks are worth stashing in a local variable. But I hope they would not hurt performance. They should be a trivial check and one you would hope that the jvm could optimize. I would prefer that large methods like this at some point be refactored into smaller methods be extracting chunk of code from them. Like these 20 lines of code could be a single line method call to a method that will log the no recipients message.
So if we do that, and need to save the log level checks into a local var, then we need to pass that local var to all these methods. I know Kirk did some benchmarking of these checks back when log4j was added. We could check with him and see if he knows if these checks are worth computing once in performance critical code. Given that this method distributes to other members over the network I doubt that it would make a difference. Do you think the performance team should have a spike to look into how costly these checks are?

Reply:
I think it is okay we merged this particular change in since it does not add more checks; just inverts the order of the check here with the call of getNormalDistributionManagerIds